DependencyResolution/PostSharp.Samples.DependencyResolution.GlobalServiceContainer/Program.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.ComponentModel.Composition.Hosting;
+using System.ComponentModel.Composition.Primitives;
+using System.Reflection;
+using PostSharp.Aspects;
+using PostSharp.Extensibility;
+using PostSharp.Serialization;
+using System.IO;
+
+
+/// <summary>
+/// Using a Global Composition Container
+/// see code samples at <see href="https://doc.postsharp.net/global-service-container"/>
+/// </summary>
+namespace PostSharp.Samples.DependencyResolution.GlobalServiceContainer
+{
+    public interface ILogger
+    {
+        void Log(string message);
+    }
+
+    public static class AspectServiceInjector
+    {
+        /// <summary>
+        /// A class instance of a Managed Extensibility Framework Container.
+        /// The Managed Extensibility Framework or MEF is a library for creating 
+        /// lightweight, and extensible applications. It allows application 
+        /// developers to discover and use extensions with no configuration required.
+        /// See the following link for more about MEF, 
+        /// <see href="https://docs.microsoft.com/en-us/dotnet/framework/mef/"/>.
+        /// </summary>
+        private static CompositionContainer container; //
+
+        public static void Initialize(ComposablePartCatalog catalog)
+        {
+            container = new CompositionContainer(catalog);
+        }
+
+        public static void BuildObject(object o)
+        {
+            if (container == null)
+                throw new InvalidOperationException();
```

Comment:
I think you need a better mechanism to cope with the situation that the container has not been initialized because the order of initialization is hard to predict. You may add unsuccessful requests to a ConcurrentQueue of initialization requests, which would then be initialized in Initialize. Your aspect will need to cope with the situation that it can be executed before the service initialization.